master/internal/hpimportance/hpimportance.go

```diff
@@ -1,15 +1,227 @@
 package hpimportance
 
+/**
+This file computes the HP importance for the HP visualizations.
+It uses the CloudForest utility (github.com/ryanbressler/CloudForest).
+
+The core steps are create the data file, run growforest which outputs
+the importance, read and return the values.
+**/
+
 import (
+	"encoding/csv"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path"
+	"sort"
+	"strconv"
+
 	"github.com/determined-ai/determined/master/pkg/model"
 )
 
-func computeHPImportance(data []model.HPImportanceTrialData) map[string]float64 {
-	// This is a place-holder for a future Random Forest implementation to actually compute the
-	// importance of each hyperparameter.
-	output := make(map[string]float64)
-	for key := range data[0].Hparams {
-		output[key] = 0.0
+const (
+	maxDiffCompBatch = 4
+	minNumberTrials  = 50
+	idealBatchDiff   = 2
+	nReplications    = 2
+
+	arffFile       = "data.arff"
+	importanceFile = "importance.txt"
+)
+
+// The data needs to be put into an arff format.
+// Arff files require variable declaration at the top,
+// where order does matter. We need to keep define the Hps and
+// keep track of the order so the data columns will match.
+func createDataFile(data map[int][]model.HPImportanceTrialData,
+	experimentConfig *model.ExperimentConfig, workingDir string) (int, error) {
+	f, err := os.Create(path.Join(workingDir, arffFile))
```

Comment:
maybe just pass `path.Join(workingDir, arffFile)` in, since it gets constructed here and outside to pass to the `growforest` cmd, and changing one without the other would be a bug.